libs/vkd3d/command.c

```diff
@@ -2675,41 +2681,25 @@ static void d3d12_command_list_update_descriptor_table(struct d3d12_command_list
 
         for (j = 0; j < range->descriptor_count; ++j, ++descriptor)
         {
-            unsigned int register_idx = range->base_register_idx + j;
-
-            /* Track UAV counters. */
-            if (list->state->uav_counter_mask != 0 && range->descriptor_magic == VKD3D_DESCRIPTOR_MAGIC_UAV)
-            {
-                const struct vkd3d_shader_uav_counter_binding *counter_bindings = list->state->uav_counters;
-                for (k = 0; k < VKD3D_SHADER_MAX_UNORDERED_ACCESS_VIEWS; k++)
-                {
-                    if (list->state->uav_counter_mask & (1u << k))
-                    {
-                        if (counter_bindings->register_space == range->register_space &&
-                            counter_bindings->register_index == register_idx)
-                        {
-                            VkBufferView vk_counter_view = descriptor->magic == VKD3D_DESCRIPTOR_MAGIC_UAV
-                                                           ? descriptor->u.view->vk_counter_view : VK_NULL_HANDLE;
-                            if (bindings->vk_uav_counter_views[k] != vk_counter_view)
-                                bindings->uav_counter_dirty_mask |= 1u << k;
-                            bindings->vk_uav_counter_views[k] = vk_counter_view;
-                            break;
-                        }
-                        counter_bindings++;
-                    }
-                }
-            }
-
             if (!vk_write_descriptor_set_from_d3d12_desc(current_descriptor_write,
                     current_image_info, descriptor, range->descriptor_magic,
-                    bindings->descriptor_set, range->binding, j))
+                    bindings->descriptor_set, range->binding, j, false))
                 continue;
 
             ++descriptor_count;
             ++current_descriptor_write;
             ++current_image_info;
 
-            if (descriptor_count == ARRAY_SIZE(descriptor_writes))
+            if (vk_write_descriptor_set_from_d3d12_desc(current_descriptor_write,
+                    current_image_info, descriptor, range->descriptor_magic,
+                    bindings->descriptor_set, range->binding, j, true))
+            {
+                ++descriptor_count;
+                ++current_descriptor_write;
+                ++current_image_info;
+            }
+
+            if (descriptor_count >= ARRAY_SIZE(descriptor_writes) - 1)
```

Comment:
I suppose this takes care of the overflow, would be good with a comment to signal intent.

Reply:
Sure, will do.